pkg/core/diagnostics.go

```diff
@@ -31,16 +33,19 @@ func SummarizeDiagnostics(diags diag.Diagnostics) DiagnosticsSummary {
 	}
 	for _, d := range diags {
 		summary.Total++
-		if _, ok := summary.BySeverity[d.Severity().String()]; ok {
-			summary.BySeverity[d.Severity().String()] = 1
+		severity := strings.ToLower(d.Severity().String())
+		dtype := strings.ToLower(d.Type().String())
+
+		if _, ok := summary.BySeverity[severity]; ok {
+			summary.BySeverity[severity] = summary.BySeverity[severity] + 1
```

Comment:
This rule is more about consistency across our codebases (not correctness).
Happy to change it

Reply:
No worries fixed it